baker/updateAvailableEntities.ts

```diff
@@ -0,0 +1,212 @@
+        await pMap(
+            commonVariables,
+            async ({ variableId, useCount }) => {
+                const variableData = await getVariableData(variableId)
+                console.log(
+                    `Pre-fetched variable ${variableId}: ${variableData.metadata.name} (${useCount} uses)`
+                )
+                return [variableId, variableData]
+            },
+            { concurrency: FETCH_CONCURRENCY }
+        )
+    )
+}
+
+const getVariableDataUsingCache = async (
+    variableId: number
+): Promise<OwidVariableDataMetadataDimensions> => {
+    if (_commonVariablesMap?.has(variableId)) {
+        _fetchVariablesCounters.cached++
+        return _commonVariablesMap.get(variableId)!
+    }
+
+    _fetchVariablesCounters.fetched++
+    return getVariableData(variableId)
+}
+
+const obtainAvailableEntitiesForGrapherConfig = async (
+    grapherConfig: GrapherInterface
+) => {
+    const grapher = new Grapher({ ...grapherConfig, manuallyProvideData: true })
+
+    // Manually fetch data for grapher, so we can employ caching
+    const variableIds = uniq(grapher.dimensions.map((d) => d.variableId))
+    const variableData: MultipleOwidVariableDataDimensionsMap = new Map(
+        await pMap(variableIds, async (variableId) => [
+            variableId,
+            await getVariableDataUsingCache(variableId),
+        ])
+    )
+    grapher.receiveOwidData(variableData)
+
+    // If the grapher has a chart tab, then the available entities there are the "most interesting" ones to us
+    if (grapher.hasChartTab) {
+        grapher.tab = GrapherTabOption.chart
+
+        // If the grapher allows for changing or multi-selecting entities, then let's index all entities the
+        // user can choose from. Otherwise, we'll just use the default-selected entities.
+        const canChangeEntities =
+            grapher.canChangeEntity || grapher.canSelectMultipleEntities
+
+        if (canChangeEntities)
+            return grapher.tableForSelection.availableEntityNames as string[]
+        else return grapher.selectedEntityNames
+    } else if (grapher.hasMapTab) {
+        grapher.tab = GrapherTabOption.map
+        // On a map tab, tableAfterAuthorTimelineAndActiveChartTransform contains all
+        // mappable entities for which data is available
+        return grapher.tableAfterAuthorTimelineAndActiveChartTransform
+            .availableEntityNames as string[]
+    } else return []
+}
+
+// The `entities` table has a 1-to-1 mapping between entity names and entity ids.
+// This function returns a map from entity names to entity ids, so we can easily convert name to ID.
+const obtainEntityNameToIdMap = async (trx: db.KnexReadonlyTransaction) => {
+    const entityNameToIdMap = new Map<string, number>()
+    const entities = await trx("entities").select("id", "name").stream()
+    for await (const entity of entities)
+        entityNameToIdMap.set(entity.name, entity.id)
+
+    return entityNameToIdMap
+}
```

Comment:
@danyx23 Does it make sense to move this method into a `db/model/Entity` file, which doesn't currently exist? We will also need the reverse mapping (i.e. id-to-name) in the future.